### PR TITLE
Document enable_database_recovery

### LINKED ISSUE #284: Configuration option `[couchdb] enable_database_recovery` is not documented
The configuration option `[couchdb] enable_database_recovery` is not documented. It is used in CouchDB [here](https://github.com/apache/couchdb/blob/f999071ec6b347b27526047cc16a00811e21fdd9/src/couch/src/couch_file.erl#L236-L248).

The configuration option, when set to true, causes database deletion to rename the database file rather than actually delete it, allowing an admin to restore the database from the backup file.

### PR DESCRIPTION
**Pull request by willholley**
## Overview

Adds documentation for couchdb.enable_database_recovery. 

## GitHub issue number

Fixes #284 

## Related Pull Requests

Description lifted from https://github.com/apache/couchdb/pull/1283

## Checklist

- [ ] Documentation is written and is accurate;
- [ ] `make check` passes with no errors
- [ ] Update [rebar.config.script](https://github.com/apache/couchdb/blob/master/rebar.config.script) with the commit hash once this PR is rebased and merged

### CODE REVIEW

File: src/config/couchdb.rst [hunk 1]
```diff
@@ -194,3 +194,16 @@ Base CouchDB Options
...
            it is advisable to review those settings and adjust them accordingly.
+
+    .. config:option:: enable_database_recovery :: Enable database recovery
+
+        Enable this to only "soft-delete" databases when
+        :ref:`DELETE /{db} <api/db>` DELETE  requests are made. This will place
+        a ``.recovery`` directory in your data directory and move deleted
+        databases/shards there instead. You can then manually delete these
+        files later, as desired.
+
```
Comment (jiangphcn) [inline src/config/couchdb.rst]: The same for full text index, i.e.  index generated by dreyfys/clouseau. The directory containing full text index will be renamed from `foo.1234567890` to `foo.20170912.092828.deleted.1234567890` for recovery.

Comment (willholley) [reply src/config/couchdb.rst]: @jiangphcn just to clarify does `enable_database_recovery` toggles index soft deletion as well or are they configured independently? Do views follow the same behavior?

Comment (flimzy) [reply src/config/couchdb.rst]: What happens in the case of deleting multiple databases with the same name?  i.e. create "foo", soft-delete "foo", re-create "foo", soft-delete "foo".

Is the original "foo" lost, or are these disambiguated in some way?

Comment (jiangphcn) [reply src/config/couchdb.rst]: @willholley one setting for all, at least for full text index https://github.com/apache/couchdb/blob/0155bd884bd8c455cc4057811cf67035b04c427f/src/dreyfus/src/dreyfus_index_manager.erl#L122. Views follow the same behavior.

Comment (willholley) [reply src/config/couchdb.rst]: > What happens in the case of deleting multiple databases with the same name? i.e. create "foo", soft-delete "foo", re-create "foo", soft-delete "foo".
good question and I don't know - I just copied the description from default.ini

Comment (jiangphcn) [reply src/config/couchdb.rst]: there will be timestamp to distinguish them with the same database name, such as `foo.20170912.092828.deleted.1234567890`.

**Pull request merged**